dam/loans/views.py

```diff
@@ -52,9 +53,10 @@ def returns(request, loan_id):
 
 def allres(request):
     res = ItemReservation.objects.filter(is_active=True)
-
-    args = {'reserves': res}
-    return render(request, 'loans/allReservations.html', args)
+    query = request.GET.get('q')
+    if query is not None:
+        res = res.filter(Q(item__name__icontains=query) | Q(item__description__icontains=query) | Q(client__first_name__icontains=query) | Q(client__last_name__icontains=query) | Q(client__email__icontains=query))
```

Comment:
Could you break this into multiple lines so this is easier to read (i.e. without needing to scroll as much)?
```suggestion
        res = res.filter(
            Q(item__name__icontains=query)
            | Q(item__description__icontains=query)
            | Q(client__first_name__icontains=query)
            | Q(client__last_name__icontains=query)
            | Q(client__email__icontains=query)
        )
```